### PR TITLE
T7/player

### PR DESCRIPTION
**Pull request by MonikaGajda**
Please look if implementation of _Cargo* getCargo(size_t index) const is_ ok for all.

### CODE REVIEW

File: shm/player.hpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class Player {
+public:
+    size_t getSpeed() const { return ship_->getSpeed(); }
+    Cargo* getCargo(size_t index) const;
+
+private:
+    std::shared_ptr<Ship> ship_;
+    size_t money_;
+    size_t availableSpace_;
+};
```
Comment (kuchichan) [inline shm/player.hpp:19]: EOF, I have preprare some .clang-file - I can put it into project :)

File: shm/player.hpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#include "cargo.hpp"
+#include "ship.hpp"
+
+class Player {
+public:
+    size_t getSpeed() const { return ship_->getSpeed(); }
+    Cargo* getCargo(size_t index) const;
+
+private:
+    std::shared_ptr<Ship> ship_;
```
Comment (kuchichan) [inline shm/player.hpp]: What about unique pointer? (Based on readme.md)

Comment (MonikaGajda) [reply shm/player.hpp]: I have updated it in new commit

**Review by kuchichan — APPROVED** [review]
Looks fine to me.

File: shm/player.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    size_t getAvailableSpace_() const { return availableSpace_; }
+    size_t getMoney() const { return money_; }
+    size_t getSpeed() const { return ship_->getSpeed(); }
+    Cargo* getCargo(size_t index) const;
+
+private:
+    std::unique_ptr<Ship> ship_;
+    size_t money_;
+    size_t availableSpace_;
+};
```
Comment (Kuba-Wi) [inline shm/player.hpp:19]: New line

**Pull request merged**